Start community category styles

MaggieCabrera opened the pull request:
Partially addresses #148

This adds some starting styles to the Community category page

Desktop:

![Screenshot 2022-01-14 at 12-43-12 Community – wporg-news-2021](https://user-images.githubusercontent.com/3593343/149510378-4f41e6b2-0ed5-4128-a1be-a3b6125939d6.png)

Mobile:

![Screenshot 2022-01-14 at 12-43-20 Community – wporg-news-2021](https://user-images.githubusercontent.com/3593343/149510370-bfb133b0-8e89-45c9-9bd4-25b1e90f4e50.png)

ryelle commented:
It looks like this template has specific hover/focus styles:
<img width="553" alt="Screen Shot 2022-01-14 at 11 56 06 AM" src="https://user-images.githubusercontent.com/541093/149554383-a31c5262-bf23-4fa2-8f6d-5df2d3326907.png">
<img width="524" alt="Screen Shot 2022-01-14 at 11 56 17 AM" src="https://user-images.githubusercontent.com/541093/149554388-6259cf08-6dfe-49df-88d2-d134a8ddddfa.png">

MaggieCabrera commented:
> It looks like this template has specific hover/focus styles: 

Oh I missed those, thank you!

ryelle commented on source/wp-content/themes/wporg-news-2021/sass/components/_categories.scss:
```diff
@@ -88,33 +88,93 @@ body.category-community {
+		&.has-post-thumbnail .entry-meta {
+
+			@include break-small() {
+				display: none;
```
Using `display:none` takes this content off the page for screen reader users too - you should be able to use `@include hide-accessibly;` instead (and then you can use the `show-hidden-accessibly` to unhide on hover/focus).

MaggieCabrera replied:
I used opacity for the hover effects in the end, I think that should be fine for a11y?

ryelle replied:
That works! Opacity does not remove the element like display:none does 👍🏻

beafialho commented:
@MaggieCabrera it's looking good!

- The black color on hover seems darker in the screenshot. Just wanted to make sure it's set to `#1E1E1E` as intended.
<img width="597" alt="Captura de ecrã 2022-01-17, às 17 24 48" src="https://user-images.githubusercontent.com/65220155/149814218-1f7776f3-5604-4d32-86d8-fa8286a4fe75.png">

- Could you please make sure that the post titles with images have the same spacing as the ones without images? Those are sitting a little bit lower than the others:
<img width="788" alt="Captura de ecrã 2022-01-17, às 17 28 23" src="https://user-images.githubusercontent.com/65220155/149814619-efdc580c-2dd9-4dff-93f4-8358f1c2c840.png">

- For tablet sizes, the titles overlap and aren't readable. Is there any chance to improve their responsiveness or is this a Gutenberg issue?

https://user-images.githubusercontent.com/65220155/149815125-af6a7686-5ff5-4f16-a995-39f5b3068346.mov

ryelle commented on source/wp-content/themes/wporg-news-2021/block-template-parts/content-category-community.html:
```diff
@@ -1,10 +1,10 @@
 
 	<!-- wp:group {"className":"entry-meta"} -->
 	<div class="wp-block-group entry-meta">
+		<!-- wp:post-title {"level":3,"isLink":true} /-->
```
The parent heading here is `h1`, so this should be an `h2` to avoid skipping levels ([more info on why here](https://webaim.org/techniques/semanticstructure/#headings)).

<img width="310" alt="Screen Shot 2022-01-17 at 12 44 53 PM" src="https://user-images.githubusercontent.com/541093/149816970-857f6f1b-45b1-4b95-b956-9441a70e0d01.png">

ryelle commented on source/wp-content/themes/wporg-news-2021/sass/components/_categories.scss:
```diff
@@ -88,33 +88,131 @@ body.category-community {
+			}
+		}
+		// Featured image only for posts that have one
+		&.has-post-thumbnail {
```
Can this ruleset be merged with the `&.has-post-thumbnail {` above (on line 130)?

ryelle commented on source/wp-content/themes/wporg-news-2021/sass/components/_categories.scss:
```diff
@@ -88,33 +88,131 @@ body.category-community {
+			position: relative;
+
+			@include break-small() {
+				&:hover {
```
This shows up correctly on hover, but not when tabbing through the page with the keyboard. You should be able to use `focus-within` to also show the content when the links are focused, ex:

```suggestion
				&:hover,
				&:focus-within {
```

ryelle reviewed (changes requested):
This is looking good! I have a few small changes requested, and could you run the linter & fix the issues?

```
yarn workspace wporg-news-2021-theme lint:css --fix
```

MaggieCabrera commented:
@beafialho is it ok if we jump from 4 per column to 3 or 2 when we go to tablet size to avoid the issue you highlighted?

beafialho commented:
@MaggieCabrera yes, we can use 3 or 2 columns

MaggieCabrera commented:
ok, this is ready for another review, I fixed all the mentioned issues, and changed the grid to drop to 3, then 2 then 1 column while we lower the viewport width

beafialho commented:
Looks perfect @MaggieCabrera!

Do you have any idea why this post has the off white background?

<img width="1316" alt="Captura de ecrã 2022-01-18, às 09 57 38" src="https://user-images.githubusercontent.com/65220155/149914636-0042176f-57e1-492d-bd40-39ffac3ee298.png">

MaggieCabrera commented:
> Looks perfect @MaggieCabrera!
> 
> Do you have any idea why this post has the off white background?
> 

It's marked with the class `.has-post-thumbnail` so it should be showing an image instead of the blue background, I'm not sure if there's a problem with the imported content or what's going on there. I could add the blue background always by default so this is not a problem

MaggieCabrera commented:
> It's marked with the class `.has-post-thumbnail` so it should be showing an image instead of the blue background, I'm not sure if there's a problem with the imported content or what's going on there. I could add the blue background always by default so this is not a problem

Nevermind that, I did that but the text only appears on hover, as do all the rest of the posts with an image. I think it's best to just make sure that the post has a thumbnail correctly in this case

ryelle commented:
I have the same post missing a featured image, I think there must've been an issue in the export. If I remove the (broken) featured image, it works as expected.

ryelle reviewed (approved):
👍🏻  
Thanks for all your work!
